crates/circuit/src/circuit_data.rs

```diff
@@ -1960,6 +2008,144 @@ impl CircuitData {
+            let remapped_qubits: Vec<Qubit> = match qubit_map {
+                Some(qubit_map) => other
+                    .get_qargs(inst.qubits)
+                    .iter()
+                    .map(|q| qubit_map[q.index()])
+                    .collect(),
+                None => other.get_qargs(inst.qubits).to_vec(),
+            };
+            let remapped_clbits: Vec<Clbit> = other.get_cargs(inst.clbits).to_vec();
+
+            let qubits = self.qargs_interner.insert(&remapped_qubits);
+            let clbits = self.cargs_interner.insert(&remapped_clbits);
+
+            self.data.push(PackedInstruction {
+                op: inst.op.clone(),
+                qubits,
+                clbits,
+                params: inst.params.clone(),
+                label: None,
+                #[cfg(feature = "cache_pygates")]
+                py_op: OnceLock::new(),
+            });
+        }
+
+        self.add_global_phase(other.global_phase())?;
+        Ok(())
+    }
+
+    /// Construct the inverse circuit
+    pub fn inverse(&self) -> PyResult<CircuitData> {
+        let inverse_global_phase =
+            Python::with_gil(|py| -> Param { multiply_param(self.global_phase(), -1.0, py) });
+
+        let mut inverse_circuit = CircuitData::clone_empty_like(self, None)?;
+        inverse_circuit.set_global_phase(inverse_global_phase)?;
+
+        for inst in self.data.iter().rev() {
+            let inverse_inst: Option<(StandardGate, SmallVec<[Param; 3]>)> = match &inst.op.view() {
+                OperationRef::StandardGate(gate) => gate.inverse(inst.params_view()),
+                _ => None,
+            };
+
+            if inverse_inst.is_none() {
+                return Err(CircuitError::new_err(format!(
+                    "The circuit cannot be inverted: {} is not a standard gate.",
+                    inst.op.name()
+                )));
+            }
+
+            let (inverse_op, inverse_op_params) = inverse_inst.unwrap();
+            let inverse_params = (!inverse_op_params.is_empty())
+                .then(|| Box::new(inverse_op_params.iter().cloned().collect()));
+
+            inverse_circuit.data.push(PackedInstruction {
+                op: inverse_op.into(),
+                qubits: inst.qubits,
+                clbits: inst.clbits,
+                params: inverse_params,
+                label: None,
+                #[cfg(feature = "cache_pygates")]
+                py_op: OnceLock::new(),
+            });
+        }
+        Ok(inverse_circuit)
+    }
+
+    /// Constructs from the definition of a standard gate
+    pub fn from_standard_gate_definition(
+        standard_gate: StandardGate,
+        params: &[Param],
+    ) -> PyResult<Self> {
+        standard_gate
+            .definition(params)
+            .ok_or(CircuitError::new_err(format!(
+                "Error extracting the definition of a standard gate {:?}",
+                standard_gate
+            )))
+    }
```

Comment:
This may be a bit redundant, the gate itself should already provide you with its standard gate definition based on StandardGate::definiton`, I don't see why this needs to be part of the `CircuitData` API just yet.

Reply:
This is a valid point, I have removed this function in [106559b](https://github.com/Qiskit/qiskit/pull/13929/commits/106559bc68948bb1e6959e3cb008602a980f26ee)